components/builder-depot-client/src/lib.rs

```diff
@@ -187,32 +187,34 @@ pub struct OriginPrivateSigningKey {
     pub owner_id: u64,
 }
 
-#[derive(Clone, Deserialize)]
-pub struct Package {
-    pub ident: PackageIdent,
-    pub checksum: String,
-    pub manifest: String,
-    pub deps: Vec<PackageIdent>,
-    pub tdeps: Vec<PackageIdent>,
-    pub exposes: Vec<u32>,
-    pub config: String,
-}
+mod json {
+    #[derive(Clone, Deserialize)]
+    pub struct Package {
+        pub ident: PackageIdent,
+        pub checksum: String,
+        pub manifest: String,
+        pub deps: Vec<PackageIdent>,
+        pub tdeps: Vec<PackageIdent>,
+        pub exposes: Vec<u32>,
+        pub config: String,
+    }
 
-#[derive(Clone, Deserialize)]
-pub struct PackageIdent {
-    pub origin: String,
-    pub name: String,
-    pub version: String,
-    pub release: String,
-}
+    #[derive(Clone, Deserialize)]
+    pub struct PackageIdent {
```

Comment:
This actually gets us very close and this was the shape that @fnichol and I agreed as the safest for this iteration. Before, we were taking this dummy `PackageIdent` struct and handing it out to other crates. Now we just use it as an interstitial point to deserialize the data off the wire and we don't hand the dummy type out to any other crates.

Reply:
Ah, OK, cool.

Does it need to be `pub` then?

Reply:
I think so. It's public but since its an internal module I think it's only accessible internally.

Reply:
Haha, yes to all of this 😄 

It seemed for the moment that by isolating these deserializing structs into their own module that we've walled them off, have a crisp way to address them in this module, and prevent them from leaking outside.

There might even be an alternative future where our REST API was described via Swagger which would allow some serializing/deserializing structs to be auto-generated for this one and only use case (similar to a protobuf def'n). For the moment though, at least this representation of an ident is an internal implementation detail now (as opposed to before this changset)